lib/project/setup-project.js

```diff
@@ -11,7 +11,20 @@ const npmConfig = require('../npm/config');
 module.exports = async function setupProject(projectRoot) {
   const config = await npmConfig(projectRoot); // kinda slow, TODO: re-implement as standalone lib
   // const { policies } = options;
-  const pkg = JSON.parse(fs.readFileSync(`${projectRoot}/package.json`, 'utf-8'));
+  const pkgPath = `${projectRoot}/package.json`;
+  if (!fs.existsSync(pkgPath)) {
+    throw new Error(`${pkgPath} does not exist, are you sure this is a valid package?`);
+  }
+  if (!fs.statSync(pkgPath).isFile()) {
+    throw new Error(`${pkgPath} is not a file, are you sure this is a valid package?`);
+  }
+
+  let pkg;
+  try {
+    pkg = JSON.parse(fs.readFileSync(pkgPath, 'utf-8'));
+  } catch (e) {
+    throw new Error(`${pkgPath} is not a valid JSON file, are you sure this is a valid package?`);
```

Comment:
to prevent an error thrown from `fs.readFileSync` become included in the invalid JSON error message, we should likely adjust the code slightly.

There are several options, likely the simplest is combine all the checks, into something as follows:

```js
let pkg;
try {
  pkg = JSON.parse(fs.readFileSync(pkgPath, 'utf-8'));
} catch (e) {
  if (typeof e === 'object' && e !== null && e.code == 'ENOENT') {
    throw new Error(`'${pkgPath}' does not exist, are you sure this is a valid package?`);
  } else {
    throw new Error(`'${pkgPath}' is not a valid JSON file, are you sure this is a valid package?`);
  }
}
```

---

Alternatively, one could move the `fs.readFileSync` call to above try/catch

Reply:
I opted for the former change, since we manually check for on-existent paths before this point.